app/components/layout/SecondaryNavBar.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from "react";
+import { useRouter } from "next/router";
+import styled from "@emotion/styled";
+import NextLink from "next/link";
+import { FormControl } from "@mui/material";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import SelectMui, { SelectMuiOption } from "../base/SelectMui";
+import { PURPLE_023, PURPLE_075, PURPLE_100, WHITE_PRIMARY } from "../colors";
+import { ROBOTO } from "../typography";
+import { breakDown } from "../breakpoints";
+
+export const NAV_BAR_HEIGHT = 70;
+
+const ProductSelectorWrapper = styled.div`
+  margin: 0 20px;
+  padding: 0 20px;
+  width: 400px; // Same value as LEFT_SIDEBAR_WIDTH from ../../../app/components/layout/Layout.tsx
+
+  @media (${breakDown("md")}) {
+    width: 300px;
+    margin: unset;
+    padding: unset;
+  }
+`;
+
+const NavBarWrapper = styled.div`
+  display: flex;
+  width: calc(100vw - 420px);
+  padding-right: 2em;
+`;
+
+export type SecondaryNavBarProps = {
+  navItems: SelectMuiOption[];
+  selectedSecondaryNavItem: SelectMuiOption;
+  setSelectedSecondaryNavItem: Function;
+  productOptions: SelectMuiOption[];
+  selectedProduct: SelectMuiOption;
+  setSelectedProduct: Function;
+};
+
+export default function SecondaryNavBar({
+  navItems,
+  selectedSecondaryNavItem,
+  setSelectedSecondaryNavItem,
+  productOptions,
+  selectedProduct,
+  setSelectedProduct,
+}: SecondaryNavBarProps) {
+  const router = useRouter();
+
+  useEffect(() => {
+    // Update the route and page based on the selectedSecondaryNavItem
+    if (selectedSecondaryNavItem && selectedProduct) {
+      router.push(selectedSecondaryNavItem.href(selectedProduct?.value));
+    }
+  }, [selectedSecondaryNavItem, selectedProduct]);
```

Comment:
After giving it some more thought, I think this approach that we landed on might be a bit backwards. Having the state of `selectedSecondaryNavItem` control the route means that if they ever get out of sync, the user will be redirected somewhere they don't necessarily want to go

For instance, if I click "Code Samples" in the secondary nav, click "Home" in the side nav, and then scroll down and click the "SDKS & Tools" link, I wind up back on the "Code Samples" page because that's still the value of `selectedSecondaryNavItem`

Rather than having the value of `selectedSecondaryNavItem` determine the value of the router, I think it would be safer to have the current path in the router determine the value of `selectedSecondaryNavItem`

Explained another way, here is the current flow:
1. User selects a category in either the secondary nav (on desktop) or the side nav (on mobile)
2. The value of `selectedSecondaryNavItem` is updated based on their selection
3. The selected value in the side nav dropdown is updated to reflect the new value of `selectedSecondaryNavItem`
4. This `useEffect` detects that the value of `selectedSecondaryNavItem` has changed and pushes a new route based on it

And this is the approach that I think might be better:
1. User selects a category in either the secondary nav (on desktop) or the side nav (on mobile)
2. A new route is pushed based on their selection
3. A new `useEffect` detects that the current path has changed and sets a new value for `selectedSecondaryNavItem` based on the new path
4. The selected value in the side nav dropdown is updated to reflect the new value of `selectedSecondaryNavItem`

That was a rather wordy explanation so if I failed to explain my thoughts clearly, please put something on my calendar and I can try to show you what I mean!